### DISCUSSION TITLE
Silence PyMeshLab?

### DISCUSSION THREAD

**Discussion by ghost** [cnr-isti-vclab/PyMeshLab#94] [Q&A]
Hi, 

I was wondering if there is an easy way to silence `pymeshlab`. When it is imported, I see the following:
```python
import pymeshlab
PyMeshLab 0.2 based on MeshLab 2020.12d
```
While that information is helpful, it would be nice if it were optional.

Additionally, when using the u3d exporter, I get the following output:
```bash
Profile = 0
Scaling factor = 1.000000
Debug level = 0
Position Quality = 500
Texture Coordinate Quality = 1000
Normal Quality = 1000
Diffuse Color Quality = 1000
Specular Color  Quality = 1000
Geometry Default Quality = 1000
Texture Quality = 100
Animation Quality = 1000
Zero Area Faces Removal = DISABLED
Exclude Normals = TRUE
Export Option Flags = ffff
Texture size limit = 0

Converting
Material Resources (1)		|	Done
Model Resources (1)		|	Done
Nodes (1)			|	Done
Modifiers (1)			|	Done
Exit code = 0
```
Again, this is useful information but it is polluting my output stream. I know from previous discussions that PyMeshLab uses an external library (or something like that) to export u3d so perhaps that is not something you have control over.

Thanks for your help!

**Comment by alemuntoni** [comment]
Yes, I agree. 
I'll remove the output when importing pymeshlab, which will be visible just calling `pymeshlab.print_pymeshlab_version()`.

About the u3d exporter, I am actually surprised that it apparently works in some systems!
Generally speaking, all filters should be muted by default in pymeshlab, but some of them print their logs in any case, and they need to be investigate one by one...

**Reply by alemuntoni** [reply]
@bdf-imbio just out of curiosity, which platform are you using?

**Reply by ghost** [reply]
I am using pymeshlab in an Ubuntu Docker container. Initially the u3d converter didn't work (#25) but providing a locale fixed things. Since then I've been using a containerized pymeshlab to produce some u3d figures. My only issue with that aspect is that the converter appears to be printing its logs when run (which I mentioned above).
